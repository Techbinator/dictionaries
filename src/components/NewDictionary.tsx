import React from "react";
import { TaddDictionary } from "../store/Dictionaries.type";
interface INewDictionary {
  addDictionary: TaddDictionary;
}

const INITIAL_STATE = { name: "", error: "" };
export default class NewDictionary extends React.Component<INewDictionary> {
  state = INITIAL_STATE;
  onSubmit = (e: React.MouseEvent<HTMLElement>) => {
    if (this.state.name.length < 1) {
      this.setState({ error: "Please add a dictionary name" });
      return;
    }
    this.props.addDictionary(this.state.name);
    this.setState(INITIAL_STATE);
  };
  render() {
    const { name, error } = this.state;
    return (
      <tr>
        <td colSpan={2} className="p-relative">
          <input
            onChange={e => this.setState({ name: e.target.value })}
            onFocus={e => this.setState({ error: "" })}
            type="string"
            placeholder="Name"
            value={name}
          />
          {error && <span className="error">{error}</span>}
        </td>
        <td>
          <input type="button" value="Add dictionary" onClick={this.onSubmit} />
        </td>
      </tr>
    );
  }
}
