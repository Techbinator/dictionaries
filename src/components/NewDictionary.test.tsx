import React from "react";
import NewDictionary from "./NewDictionary";
import { shallow } from "enzyme";
import contextMock from "../helpers/__mocks__/context.mock";

describe("NewDictionary component", () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = shallow(
      <NewDictionary addDictionary={contextMock.addDictionary} />
    );
  });
  it("should render snapshot with default values", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should add dictionary", () => {
    wrapper.find('input[placeholder="Name"]').simulate("change", {
      target: { value: "Name" }
    });
    expect(wrapper.instance().state).toEqual({ error: "", name: "Name" });
    wrapper.find('input[type="button"]').simulate("click");
    expect(contextMock.addDictionary).toBeCalledWith("Name");
  });

  it("should throw validation error", () => {
    wrapper.find('input[placeholder="Name"]').simulate("change", {
      target: { value: "" }
    });
    wrapper.find('input[type="button"]').simulate("click");
    expect(wrapper.instance().state).toEqual({
      error: "Please add a dictionary name",
      name: ""
    });

    expect(wrapper.find(".error").text()).toEqual(
      "Please add a dictionary name"
    );
  });
});
