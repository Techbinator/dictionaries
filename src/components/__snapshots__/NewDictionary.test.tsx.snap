// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewDictionary component should render snapshot with default values 1`] = `
<tr>
  <td
    className="p-relative"
    colSpan={2}
  >
    <input
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Name"
      type="string"
      value=""
    />
  </td>
  <td>
    <input
      onClick={[Function]}
      type="button"
      value="Add dictionary"
    />
  </td>
</tr>
`;
